### ISSUE TITLE
diagnostic range is off when the line is indented with tabs

### ISSUE THREAD

**Issue by InSyncWithFoo** [astral-sh/ty#203]
Given this file:

```python
def f():
	return (1 == '2')()  # Tab indented
```

Ruff gets the range correctly:

```python
$ ruff check --select PLR0133 --isolated
foo.py:2:10: PLR0133 Two constants compared in a comparison, consider replacing `1 == '2'`
  |
1 | def f():
2 |     return (1 == '2')()  # Tab indented
  |             ^ PLR0133
  |

Found 1 error.
```

But Red Knot doesn't:

```shell
$ red_knot check --venv-path .venv
error: lint:call-non-callable
 --> ~\project\foo.py:2:9
  |
1 | def f():
2 |     return (1 == '2')()  # Tab indented
  |        ^^^^^^^^^^^^ Object of type `bool` is not callable
  |
```

**Comment by AlexWaygood** [comment]
It appears that this has been fixed by https://github.com/astral-sh/ruff/pull/18962 🥳

@BurntSushi, do you think it's worth adding another regression test for this specifically, or would that just duplicate the tests you already added in that PR?

**Comment by BurntSushi** [comment]
Nice find! Sounds like a good idea: https://github.com/astral-sh/ruff/pull/18965
